improver/utilities/solar.py

```diff
@@ -224,14 +227,30 @@ def _create_daynight_mask(self, cube):
                 will be ignored although they might appear as attributes
                 on the cube as it is extracted from the first slice.
         """
-        daynight_mask = next(
-            cube.slices(
-                [cube.coord("time"), cube.coord(axis="y"), cube.coord(axis="x")]
-            )
-        ).copy()
-        daynight_mask.rename("day_night_mask")
-        daynight_mask.units = unit.Unit("1")
-        daynight_mask.data = np.ones(daynight_mask.data.shape, dtype="int") * self.night
+        slice_coords = [cube.coord(axis="y"), cube.coord(axis="x")]
+        if cube.coord("time") in cube.coords(dim_coords=True):
+            slice_coords.insert(0, cube.coord("time"))
+
+        template = next(cube.slices(slice_coords))
+        demoted_coords = [
+            crd
+            for crd in cube.coords(dim_coords=True)
+            if crd not in template.coords(dim_coords=True)
+        ]
+        for crd in demoted_coords:
+            template.remove_coord(crd)
+        attributes = generate_mandatory_attributes([template])
+        title_attribute = {"title": "Day-Night mask"}
```

Comment:
I think most of our titles have "on the UK standard grid" or simplar, or equivalent for the global domain in our titles. I guess that is hard to put in here unless you do something clever with the title on the input cube which probably wouldn't be very robust. I think this is probably ok.

Reply:
I thought there was a metadata utility to figure this out, but I may have dreamt it as I cannot find it, so I will leave it be.